### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by Tadessejemal1**
Project requirements

- Create a class Classroom with the following:

        1. @label instance variable, should be initialized in the constructor.
        2. Setter and getter for @label (remember about attr_accessor).

- Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:

          1. Create the has-many side (a classroom has many students).
         2.  Create the belongs-to side (a student belongs to a classroom).
          3. Make sure that when adding a student to a classroom it also sets the classroom for the student.
          4. Make sure that when setting the classroom for a student it also adds it to the classrooms' students.

- Create a class Book with the following:

        1. @title and @author instance variables, should be initialized in the constructor.
        2.  Setters and getters for instance variables (remember about attr_accessor).

- Create a class Rental with the following:

         1. @date instance variable, should be initialized in the constructor.
         2.  Setter and getter for @date (remember about attr_accessor).

- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:

         1. Create the has-many side of Book and Rental (a book has many rentals).
         2. Create the belongs-to side of Rental and Book (a rental belongs to a book).
         3. Create the has-many side of Person and Rental (a person has many rentals).
         4. Create the belongs-to side of Rental and Person (a rental belongs to a person).
         5. Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+require_relative './rental'
+# mant to many relationship between person and book
+
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (rloterh) [inline book.rb]: - [ ] `rental` should be marked with `attr_accessor`, as we intend to both _access (get)_ as well as _modify (set)_ it. Kindly ensure to improve upon this aspect.

File: person.rb [hunk 2]
```diff
@@ -1,7 +1,8 @@
 require_relative './nameable'
+require_relative './rental'
 
 class Person < Nameable
-  attr_reader :id, :parent_permission
+  attr_reader :id, :parent_permission, :rentals
```
Comment (rloterh) [inline person.rb]: - [ ] `rental` should be marked with `attr_accessor`, as we intend to both _access (get)_ as well as _modify (set)_ it. Kindly ensure to improve upon this aspect.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Tadesse, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Good job implementing associations so far

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

**You can also consider:**
- I think you should change the header title _Authors_ to _Author_ in your ReadMe, as this project is worked on by a sole contributor. 
Implementing the above change(s) will make your ReadMe/ repository stand out even better 😉

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions
-N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- 

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**